src/main/java/ArrayStack.java

```diff
@@ -44,9 +46,46 @@ public void push(T element) {
     @SuppressWarnings("unchecked")
     private void expandCapacity() {
         T[] newStack = (T[]) new Object[stack.length * 2];
-        for (int i = 0; i < stack.length; ++i) {
-            newStack[i] = stack[i];
-        }
+        System.arraycopy(stack, 0, newStack, 0, stack.length);
```

Comment:
This was deliberate because:
- I want them writing the expand capacity for learning purposes as this stage, and `copyOf` can do everything
- Was going to replace the whole `expandCapacity` function for next time

Reply:
Agreed.